### ISSUE TITLE
`TopicReference` should inherit from `TokenizedTopicList`

### ISSUE THREAD

**Issue by JeremyCaney** [OnTopicCMS/OnTopic-Editor-AspNetCore#27]
The `TopicReference` attribute type should inherit from the `TokenizedTopicList ` base type in order to allow, optionally, the `AttributeKey` and `AttributeValue` filtering (from `QueryableTopicList`) and `ResultLimit` and `AuroPostBack` (from `TokenizedTopicList`.

Additionally, it should expose a new `UseCurrentContentType` option in order to support the `BaseTopic` functionality. 

This allows it to continue supporting the special `BaseTopic` use case, while simultaneously supporting more general applications of topic references.

**Comment by JeremyCaney** [comment]
As an alternative to exposing a `UseCurrentContentType`, we could do one of the following:
- Expose a `IsBaseTopic`, which automatically triggers certain properties.
- Hard-code special rules into the view component if the `AttributeKey` is `BaseTopic`.
- Introduce a new attribute type, `BaseTopic`, which is used exclusively for `BaseTopic`. 

I don’t love any of these. But they might make more sense than exposing multiple attributes exclusively intended for `BaseTopic` to the configuration of every `TopicReference`.

**Comment by JeremyCaney** [comment]
This dramatically simplified the implementation of `TopicReferenceViewComponent` while making it more flexible by deriving `TopicReferenceAttributeDescriptorViewModel` from `TokenizedTopicListAttributeDescriptorViewModel`, and thus sharing its `ResultLimit` and `RootTopic` properties, while also allowing implementers to specify whether they want `AutoPostBack`, and even to customize the filtering via `AttributeKey` and `AttributeValue` (4319cd0). 

Critically, this allows us to customize the `TopicReferenceAttributeDescriptorViewModel` directly in the `TopicReferenceViewComponent` and pass that object down to `TokenizedTopicListViewComponent`, instead of constructing a new `TokenizedTopicListAttributeDescriptorViewModel` within the view (7bba5f6, 798ff70). 

While I was at it, I introduced the new `UseCurrentContentType` property, which will allows us to configure `BaseTopic`'s functionality without any special hard-coded rules (7bdedac). 

> _Note:_ As this wasn't introduced in OnTopic Editor 5.0.0, hard-coded rules remain for backward compatibility with `BaseTopic`; those will be removed in OnTopic 6.0.0.)

This functionality was merged into `develop` as part of add1924.
